students/OwenGaustad/Lab4/TurnSignal/TurnSignal.c

```diff
@@ -0,0 +1,184 @@
+// ------- Preamble -------- //
+#include <avr/io.h>
+#include <util/delay.h>
+
+#define LED_PORT                PORTD
+#define LED_DDR                 DDRD
+
+#define BUTTON_PORT             PORTB
+#define BUTTON_PIN              PINB
+#define BUTTON_DDR              DDRB
+
+#define BUTTON                  PB0
+#define BUTTON2                 PB1
+
+#define BV(bit)               (1 << (bit))       // Mask with single bit set
+#define toggleBit(byte, bit)  (byte ^= BV(bit))  // toggle bit
+
+int offState = 0;  //This variable goes between 0 and 1 and controls when all lights turn off
+int delay;         //This is used in my for loops and is used as the integer increaded to our maximum time
+int led;           //This variable is used to determine which LED 0-7 is turned on
+int time;          //This variable is used with the method variableDelay() and is our value for waited time in milliseconds
```

Comment:
From what I can tell, these are all used strictly internally to functions. If that's the case, you want to declare them in the function rather than global. Especially `time` which is actually passed into the function... I'm suspecting this may be the root of the issue you were seeing with values changing.
Limiting the scope of the variables does two things:
1. Cuts down on variable confusion where you might accidentally use a global variable for a local task
2. saves memory because global variables must be kept while local variables are discarded when the function returns.